test/functional/config_templates_controller_test.rb

```diff
@@ -74,18 +74,36 @@ class ConfigTemplatesControllerTest < ActionController::TestCase
     assert !ConfigTemplate.exists?(config_template.id)
   end
 
-  test "build menu" do
-    template = File.read(File.expand_path(File.dirname(__FILE__) + "/../../app/views/unattended/pxe/PXELinux_default.erb"))
-    ConfigTemplate.find_by_name('PXELinux global default').update_attribute(:template, template)
-
-    ProxyAPI::TFTP.any_instance.stubs(:fetch_boot_file).returns(true)
-    Setting[:unattended_url] = "http://foreman.unattended.url"
-    @request.env['HTTP_REFERER'] = config_templates_path
-
-    ProxyAPI::TFTP.any_instance.expects(:create_default).with(has_entry(:menu, regexp_matches(/ks=http:\/\/foreman.unattended.url:80\/unattended\/template/))).returns(true)
-
-    get :build_pxe_default, {}, set_session_user
-    assert_redirected_to config_templates_path
+  context "build pxe menu" do
+    setup do
+      proxy = smart_proxies(:two)
+      proxy.features = []
+      proxy.save!
```

Comment:
Only way to avoid having this fixture TFTP proxy in the tests.  It'd be nice if the feature Smart Proxies became factories, but they're used in Taxonomy tests and not touching those :ghost:  :fearful: